spec.emu

```diff
@@ -12,162 +12,113 @@ copyright: false
+            <h1>Properties of the Iterator Constructor</h1>
+
+            <emu-clause id="sec-iterator.range">
+                <h1>Iterator.range ( _start_, _end_, _option_ )</h1>
+                <emu-alg>
+                    1. If _start_ is a Number, return ? CreateNumericRangeIterator(_start_, _end_, _option_, ~number-range~).
+                    1. If _start_ is a BigInt, return ? CreateNumericRangeIterator(_start_, _end_, _option_, ~bigint-range~).
+                    1. Throw a *TypeError* exception.
+                </emu-alg>
+            </emu-clause>
         </emu-clause>
-    </emu-clause>
-    <emu-clause id="sec-properties-of-the-numericrangeiterator-prototype-object-step">
-        <h1>%NumericRangeIteratorPrototype%.step</h1>
-        <p>%NumericRangeIteratorPrototype%.step is an accessor property with attributes { [[Enumerable]]: false, [[Configurable]]: *true*}. The [[Get]] attribute is defined as follows:</p>
-        <emu-clause id="sec-properties-of-the-numericrangeiterator-prototype-object-step-getter">
-            <h1>get %NumericRangeIteratorPrototype%.step</h1>
-            <emu-alg>
-                1. Perform ? RequireInternalSlot(*this*, [[step]]).
-                1. Return *this*.[[step]].
-            </emu-alg>
+
+        <emu-clause id="sec-numeric-range-iterator-object">
+            <h1>The <dfn>NumericRangeIterator</dfn> Object</h1>
+            <p>A NumericRangeIterator object is an iterator that yields numbers. There is not a named constructor for NumericRangeIterator objects. Instead, NumericRangeIterator objects are created by the CreateNumericRangeIterator abstract operation as needed.</p>
+
+            <emu-clause id="sec-create-numeric-range-iterator" type="abstract operation">
+                <h1>
+                    CreateNumericRangeIterator (
+                        _start_: a Number or a BigInt,
+                        _end_: an ECMAScript language value,
+                        _option_: an ECMAScript language value,
+                        _type_: ~number-range~ or ~bigint-range~
+                    )
+                </h1>
+                <dl class="header"></dl>
+                <emu-alg>
+                    1. If _type_ is ~number-range~, let _primitiveType_ be *"number"*, else let _primitiveType_ be *"bigint"*.
+                    1. Assert: Type(_start_) is _primitiveType_.
+                    1. If _type_ is ~bigint-range~, let _zero_ be *0n*<sub>ℤ</sub>, else let _zero_ be *0*<sub>ℤ</sub>.
+                    1. If _type_ is ~bigint-range~, let _one_ be *1n*<sub>ℤ</sub>, else let _one_ be *1*<sub>ℤ</sub>.
+                    1. If _end_ is not *+∞* or *-∞* and if Type(_end_) is not _primitiveType_, throw a *TypeError* exception.
+                    <emu-note type="editor">Allowing all kinds of number (number, bigint, decimals, ...) to range from a finite number to infinity.</emu-note>
+                    1. If _start_ is *+∞* or *-∞*, throw a *RangeError* exception.
```

Comment:
What about NaN? I think you want “is not finite” here.

Reply:
nice catch. current test file shows if any of `start`/`end`/`step` is `NaN`, it will be an empty range. we can discuss it in a new issue. this PR is only intended to change the API shape to make it easier to review, no range algorithm is changed.